components/board/genericlinux/gpio.go

```diff
@@ -210,38 +210,6 @@ func (pin *gpioPin) startSoftwarePWM() error {
 	return nil
 }
 
-// accurateSleep is intended to be a replacement for utils.SelectContextOrWait which wakes up
-// closer to when it's supposed to. We return whether the context is still valid (not yet
-// cancelled).
-func accurateSleep(ctx context.Context, duration time.Duration) bool {
```

Comment:
I agree that this function seems correct. 

I'd recommend testing this & logging the duration on the board that this behaves well on & the board this behaves poorly on & see if different boards are providing radically different durations.